src/header.h

```diff
@@ -1,197 +1,150 @@
+{
+
+    int id;
+    pid_t fork_id;
+    int arrival;
+    int runtime;
+    int priority;
+} process_info_t;
+
+/**
+ * scheduling_algo - Enumeration representing different scheduling algorithms
+ * @HPF: Highest Priority First
+ * @SRTN: Shortest Remaining Time Next
+ * @RR: Round Robin
+ *
+ * Description: Enumeration representing different scheduling algorithms including
+ *              Highest Priority First, Shortest Remaining Time Next, and Round Robin.
+ */
+typedef enum
+{
+    HPF = 1,
+    SRTN,
+    RR
+} scheduling_algo;
+
+/**
+ * SchedulerConfig - Structure for scheduler configuration settings.
+ * @selected_algorithm: The selected scheduling algorithm.
+ * @quantum: Quantum for time slice (if applicable).
+ *
+ * Description: Structure representing the configuration settings for the scheduler,
+ *              including the selected scheduling algorithm and quantum for time slice
+ *              (if applicable).
+ */
+typedef struct
+{
+    scheduling_algo selected_algorithm;
+    int quantum;
+    int curr_quantum;
+} SchedulerConfig;
+
+//===========================================Process=============================================//
+
+/**
+ * decrementRemainingCPUTime - Decrement the remaining CPU time of a process.
+ * @param process: Pointer to the process information structure.
+ */
+void decrementRemainingCPUTime(process_info_t *process);
+
+//=======================================Scheduler=========================================//
+
+/**
+ * getSchedulerConfigInstance - Function to get the singleton instance of SchedulerConfig.
+ * @return Pointer to the instance.
+ *
+ * Description: This function returns a pointer to the singleton instance of the SchedulerConfig
+ *              structure, ensuring that only one instance exists throughout the program.
+ */ 
+SchedulerConfig *getSchedulerConfigInstance();
+
+/**
+ * struct rprocess_s - current running process
+ * @process: Pointer to the process
+ * @priority: Value of the node priority
+ *
+ * Description: save the info of the current running process
+ */
+typedef struct rprocess_s
+{
+    process_info_t *process;
+    int priority;
+} rprocess_t;
+
+/**
+ * initializeProcesses - Initializes processes based on the number of processes to be created.
+ *
+ * @param head: Pointer to the pointer to the head of the priority queue.
+ * @param num_of_processes: Number of processes to be initialized.
+ *
+ * Description: Initializes processes, forks them, and adds them to the ready queue.
+ */
+void initializeProcesses(pqueue_t **head, int num_of_processes);
+
+/**
+ * addToReadyQueue - Adds a process to the ready queue based on the scheduling algorithm.
+ *
+ * @param head: Pointer to the pointer to the head of the priority queue.
+ * @param process: Pointer to the process to be added.
+ *
+ * Description: Adds a process to the ready queue based on the selected scheduling algorithm.
+ */
+void addToReadyQueue(pqueue_t **head, process_info_t *process);
+

```

Comment:
This functions should take a general type, as it is a general priority qeueu

Reply:
This is a function dedicated to the processes. But the `push` function takes a `void *` as an argument instead of a certain data type. Ex: ```push(head, (void *)process, 0);```